packages/@extension-kid/data-layer/__tests__/dataLayer-test.ts

```diff
@@ -0,0 +1,340 @@
+  getExtensionModule,
+  DataLayerExtensionType,
+  DataLayerExtensionInterface,
+  DataLayerType,
+  DataLayer
+} from "@extension-kid/data-layer";
+import { injectable } from "inversify";
+import gql from "graphql-tag";
+import { marbles } from "rxjs-marbles/jest";
+import { Observable } from "rxjs";
+import { take } from "rxjs/operators";
+
+function toPromise(observable: Observable<any>) {
+  return new Promise((resolve, reject) =>
+    observable.subscribe({ next: resolve, complete: resolve, error: reject })
+  );
+}
+
+const JobsSymbol = Symbol("Jobs");
+@injectable()
+class JobsExtension implements DataLayerExtensionInterface {
+  id = JobsSymbol;
+
+  getResolvers(activeExtensions: symbol[]) {
+    const hasJobs = activeExtensions.includes(TasksSymbol);
+
+    const additionalJobData = hasJobs ? { taskId: "foo-task" } : {};
+
+    return {
+      Query: {
+        jobs() {
+          return [{ id: "foo", docker: "nginx", ...additionalJobData }];
+        }
+      }
+    };
+  }
+
+  getTypeDefinitions(activeExtensions: symbol[]) {
+    const hasJobs = activeExtensions.includes(TasksSymbol);
+
+    return `
+    extend type Mutation {
+      deleteJob(id: String): Boolean
+    }
+
+    extend type Query {
+      jobs: Job
+    }
+
+    type Job {
+      id: String!
+      docker: String!
+      ${hasJobs ? "taskId: String!" : ""}
+    }
+    `;
+  }
+}
+
+const TasksSymbol = Symbol("Tasks");
+// tslint:disable-next-line
+@injectable()
+class TasksExtension implements DataLayerExtensionInterface {
+  id = TasksSymbol;
+
+  getResolvers() {
+    return {
+      Query: {
+        task(_: any, args: { id?: string }) {
+          const id = args.id || "unknown";
+
+          return {
+            id,
+            log: id + "log"
+          };
+        }
+      },
+      Mutation: {
+        deleteTask(id: string) {
+          return id === "deletable-task";
+        }
+      },
+      Job: {
+        task(parent: { id: string }) {
+          return {
+            id: parent.id + "task",
+            log: parent.id + "tasklog"
+          };
+        }
+      }
+    };
+  }
+
+  getTypeDefinitions(activeExtensions: symbol[]) {
+    const hasJobs = activeExtensions.includes(TasksSymbol);
+
+    const jobTypeExtension = `
+    extend type Job {
+      task: Task
+    }
+    `;
+
+    return `
+    extend type Query {
+      task(id: String): Task
+    }
+
+    extend type Mutation {
+      deleteTask(id: String): Boolean
+    }
+
+    type Task {
+      id: String!
+      log: String!
+    }
+
+    ${hasJobs ? jobTypeExtension : ""}
+    `;
+  }
+}
+
+describe("DataLayer", () => {
+  let container: Container;
+  beforeEach(async () => {
+    container = new Container();
+
+    const dataLayerContainerModule = dataLayerContainerModuleFactory();
+
+    container.load(dataLayerContainerModule);
+  });
+
+  it(
+    "does not error without an extension",
+    marbles(m => {
+      const dl: DataLayer = container.get<DataLayer>(DataLayerType);
```

Comment:
You only need one of `: DataLayer` or `get<DataLayer>`.

Reply:
Okay, I'll remove of one them 💪